riscv_analysis/src/lints/overlapping_function.rs

```diff
@@ -27,3 +27,93 @@ impl LintPass for OverlappingFunctionCheck {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::lints::OverlappingFunctionCheck;
+    use crate::parser::RVStringParser;
+    use crate::passes::{LintError, LintPass, Manager};
+
+    /// Compute the lints for a given input
+    fn run_pass(input: &str) -> Vec<LintError> {
+        let (nodes, error) = RVStringParser::parse_from_text(input);
+        assert_eq!(error.len(), 0);
+
+        let cfg = Manager::gen_full_cfg(nodes).unwrap(); // Need fn annotations
+        OverlappingFunctionCheck::run_single_pass_along_cfg(&cfg)
+    }
+
+    #[test]
+    fn two_overlapping_functions() {
+        // Functions `fn_a` and `fn_b` share the same `ret` instruction
+        let input = "\
+            main:                      \n\
+                li     a0, 0           \n\
+                jal    fn_a            \n\
+                jal    fn_b            \n\
+                addi   a7, zero, 10    \n\
+                ecall                  \n\
+            fn_a:                      \n\
+                addi   a0, a0, 1       \n\
+            fn_b:                      \n\
+                addi   a0, a0, 2       \n\
+                ret                    \n";
+
+        let lints = run_pass(input);
+
+        assert_eq!(lints.len(), 1);
+        assert!(matches!(&lints[0], LintError::OverlappingFunctions(..)));
```

Comment:
In addition to the lint type, I think it would be more rigorous if we also check what token/line is being annotated. A simple string comparison or line number check is good enough.

Here's an example if we keep the lint on the function entry node:
```rust
        assert!(
            matches!(&lints[0], LintError::OverlappingFunctions(node, _) if matches!(node, ParserNode::FuncEntry(entry) if entry.token.text == "addi a0 a0 2"))
        );
```
And here's an example if we change the node to a label:
```rust
        assert!(
            matches!(&lints[1], LintError::OverlappingFunctions(node, _) if matches!(node, ParserNode::Label(label) if label.token.text == "fn_c:"))
        );
```